nomad/structs/structs.go

```diff
@@ -4836,6 +4912,26 @@ type DeploymentStatusUpdate struct {
 	StatusDescription string
 }
 
+type RescheduleTracker struct {
```

Comment:
I feel like this should become `RescheduleEvent` and then you embed these in a tracker.

```
type RescheduleEvent struct { /* current stuff */ }

type RescheduleTracker struct {
  Events []*RescheduleEvent
}
```

Gives us future flexibility